cse/WeightCalculation.py

```diff
@@ -32,10 +32,19 @@ def missingTermWeight():
 
 
 def cosineSimilarity(docWeights, queryWeights):
+    cache = NormCache()
```

Comment:
How sure are you that the NormCache Singleton isn't garbage collected between 2 calls of this function?